common/lib/kbuy.ts

```diff
@@ -0,0 +1,64 @@
+import type { JsonRpcSigner, TransactionResponse, TransactionReceipt } from "@ethersproject/providers";
+
+import { collectionGetContract } from "./kcollection-get";
+import { getNetwork } from "./kconfig";
+
+import { BigNumber, ethers } from "ethers";
+
+const buyNftResponse = async (
+  chainId: number,
+  address: string,
+  tokenID: string,
+  buyer: JsonRpcSigner,
+  nftPrice: string
+): Promise<TransactionResponse | null> => {
+  // console.log("transferNftResponse", chainId, address, tokenID, destinationAddress);
+
+  let txResp: TransactionResponse | null = null;
+  const network = getNetwork(chainId);
+
+  if (!(chainId && address && tokenID && network && buyer)) return txResp;
+
+  const { contract, supports } = await collectionGetContract(chainId, address, buyer.provider);
+  const connectedContract = contract.connect(buyer);
+
+  if (supports.IOpenNFTsV4) {
+    const buyOpenNftV4 = "buy(uint256)";
+    const buyFunction = connectedContract[buyOpenNftV4] as {
```

Comment:
save one line (for the planet ;-)
`const buyFunction = connectedContract["buy(uint256)"] as {`